recirq/otoc/otoc_example.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example for measuring OTOC of several random circuits."""
+
+import cirq
+import numpy as np
+from matplotlib import pyplot as plt
+
+from recirq.otoc.otoc_circuits import build_otoc_circuits
+
+
+def main():
+    # Specify project ID and processor name.
+    project_id = ''
+    engine = cirq.google.Engine(project_id=project_id,
+                                proto_version=cirq.google.ProtoVersion.V2)
+    processor_name = 'rainbow'
+
+    # Specify qubits to measure. The qubits must form a line. The ancilla qubit
+    # must be connected to the first qubit.
+    qubit_locs = [(3, 3), (2, 3), (2, 4), (2, 5), (1, 5), (0, 5), (0, 6),
+                  (1, 6)]
+    qubits = [cirq.GridQubit(*idx) for idx in qubit_locs]
+    ancilla_loc = (3, 2)
+    ancilla = cirq.GridQubit(*ancilla_loc)
+    num_qubits = len(qubits)
+
+    # Specify how the qubit interact. For the 1D chain in this example, only two
+    # layers of two-qubit gates (CZ is used here) is needed to enact all
+    # interactions.
+    int_sets = [
+        {(qubits[i], qubits[i + 1]) for i in range(0, num_qubits - 1, 2)},
+        {(qubits[i], qubits[i + 1]) for i in range(1, num_qubits - 1, 2)}]
+
+    forward_ops = {(qubit_locs[i], qubit_locs[i + 1]): cirq.Circuit([cirq.CZ(
+        qubits[i], qubits[i + 1])]) for i in range(num_qubits - 1)}
+
+    reverse_ops = {(qubit_locs[i], qubit_locs[i + 1]): cirq.Circuit([cirq.CZ(
+        qubits[i], qubits[i + 1])]) for i in range(num_qubits - 1)}
+
+    # Build two random circuit instances (each having 12 cycles).
+    circuit_list = []
+    cycles = range(12)
+    num_trials = 2
+    for i in range(num_trials):
+        np.random.seed(i)
```

Comment:
why are you setting the seed inside the loop? Consider doing it once outside the loop. For this example, it might be most appropriate to set the seed globally at the top of the file.

Reply:
Changed.